frontend/gateway/gateway.go

```diff
@@ -350,20 +350,9 @@ func (lbf *llbBridgeForwarder) Discard() {
 		delete(lbf.workerRefByID, id)
 	}
 	for id, r := range lbf.refs {
-		if lbf.err == nil && lbf.result != nil {
-			keep := false
-			lbf.result.EachRef(func(r2 solver.ResultProxy) error {
-				if r == r2 {
-					keep = true
-				}
-				return nil
-			})
-			if keep {
-				continue
-			}
-		}
 		r.Release(context.TODO())
```

Comment:
If `lbf.err` is not nil then all `lbf.result` fields need to be released here as well?

Reply:
Good catch :smile:

Have fixed this up to release the `FrontendResult` if we encounter an error, and also applied the same check in the forwarder, which means we do actually need to keep track of the cloned result proxies, so we can tidy them up in case of an error.